### DISCUSSION TITLE
URIError: URI malformed when using "jbrowse text-index"

### DISCUSSION THREAD

**Discussion by abjanssen** [GMOD/jbrowse-components#4268] [Q&A]
Dear JBrowse2 users,

I've been trying to get our JBrowse2 session up and running, but I'm running into troubles when making the search files.

I'm using `jbrowse text-edit` to index the tracks, but I'm getting the same error whatever I do (specifying specific tracks).
The command runs up to a certain percentage, and then throws an error " URIError: URI malformed".

I cannot find much about this error online, so instead I'm asking here.

If you require any files to replicate this, please let me know, I can share them if needed.

**Comment by cmdcolin** [comment]
can you paste a copy of your config.json by chance?

**Reply by abjanssen** [reply]
I have attached my config.json here. 

[config.json](https://github.com/GMOD/jbrowse-components/files/14525617/config.json)

**Reply by cmdcolin** [reply]
actually you don't necessarily need to paste your config.json. it is actually likely a specific GFF3 data entry (column 9 is supposed to be uri or "perecent  encoded" for certain characters, but potentially your files might not be following it. if you search your GFF3 files for the "%" character you might find the issue
https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md#description-of-the-format

we might be able to make a workaround in the jbrowse text indexing for this  to avoid a total crash, but it could be worth looking at your files to see if there are any percent encoding issues

**Reply by cmdcolin** [reply]
and to add on, you can likely fix it on your end if you find the potentially mis-encoded gff3 data to fix it (change it to the percent encoding for example)

**Comment by abjanssen** [comment]
So it turns out we had a rough "2%C" instead of "%2C" in our manually curated annotation file somewhere.
Thank you for your quick help, and helping me understand this error!
